### DISCUSSION TITLE
Access current locale in server directory

### DISCUSSION THREAD

**Discussion by damvik** [nuxt-modules/i18n#2010] [General]
How to access current locale in `server/api` directory? 

```js
export default defineEventHandler(async (event) => {
  const config = useRuntimeConfig();

  const { results, pagination } = await $fetch("blog/articles", {
    baseURL: config.public.apiBaseUrl,
    query: getQuery(event),
  });

  return { results, pagination };
});

```

**Comment by damvik** [comment]
Link to discussion about extending useFetch: 

- https://github.com/nuxt/nuxt/discussions/20392

There is an example how to use current locale in each request. What do you think about this approach?

**Comment by DavidDeSloovere** [comment]
With the new experimental `defineI18nLocaleDetector` we can now get translations in API routes: https://i18n.nuxtjs.org/guide/server-side-translations

I have my own middleware where I need the locale to do something.

The follow code works (although Typescript has issue, see below). 
Maybe there's another/better way to do this, suggestions are welcome.

```
export default defineEventHandler(async (event) => {
  const locale = await event.context.i18n?.locale();
  console.log("🌿🌿 event.context.i18n?.locale ", locale);

}
```


The Typescript issue:
![image](https://github.com/nuxt-modules/i18n/assets/352626/74ea2901-9579-47a5-abcb-143f084acbab)
